koku/api/iam/test/tests_customer_view.py

```diff
@@ -141,15 +174,44 @@ def test_delete_customer(self):
         # Sanity check that the customer count was reduced by 1
         self.assertEquals(Customer.objects.all().count(), customer_count - 1)
 
-    def test_delete_customer_with_reg_users(self):
-        """Test deleting customers with regular users."""
+    def test_delete_customer_with_reg_users_providers(self):
+        """Test deleting customers with regular users and providers."""
         customers = self.customers
         self.assertIsNotNone(customers)
 
         self.assertEquals(len(customers), 2)
         customer_1 = customers[0]
         customer_2 = customers[1]
 
+        # Setup Providers
+        iam_arn = 'arn:aws:s3:::my_s3_bucket'
+        bucket_name = 'my_s3_bucket'
+        token = customer_1['owner']['token']
+        response = self._create_provider(bucket_name, iam_arn, token)
+        self.assertEqual(response.status_code, 201)
+
+        bucket2_name = 'my_other_s3_bucket'
+        response = self._create_provider(bucket2_name, iam_arn, token)
+        self.assertEqual(response.status_code, 201)
```

Comment:
I think we need to fail when the same ARN is used for different providers. ARN should be unique. I just looked at the model, and realized it does not have a unique constraint on it, and it probably should have it. I can make an issue for that to handle elsewhere.

Reply:
Yes, we discussed offline that this should be addressed in a separate change.  I'll update the test case to be more realistic from the user perspective.